client/packages/system/src/Stock/api/hooks/useRepack.ts

```diff
@@ -0,0 +1,91 @@
+import { useMutation, useQuery } from '@openmsupply-client/common';
+import { useStockGraphQL } from '../useStockGraphQL';
+import { useState } from 'react';
+import { Repack } from '../../types';
+import { REPACK_LIST, STOCK_LINE } from './keys';
+
+export const useRepack = (invoiceId: string) => {
+  const { stockApi, storeId } = useStockGraphQL();
+
+  const queryFn = async () => {
+    const result = await stockApi.repack({
+      storeId,
+      invoiceId,
+    });
+
+    if (result.repack.__typename === 'RepackNode') {
+      return result.repack;
+    }
+  };
+
+  const query = useQuery({
+    queryKey: ['stock', invoiceId],
+    queryFn,
+    enabled: invoiceId !== '',
+  });
+
+  return query;
+};
+
+export const useRepackEdit = (init: Repack) => {
+  const { stockApi, storeId, queryClient } = useStockGraphQL();
+  const [repack, setRepack] = useState<Repack>({ ...init });
+  const stockLineId = repack.stockLineId ?? '';
+
+  // FETCH
+  const queryFn = async () => {
+    const result = await stockApi.repacksByStockLine({
```

Comment:
If this is an intended pattern, it might be nice to add some kind of context comment here... how come individual repack line queried in a separate hook, but the repacks list is queried in an edit hook? Intuitively I would have expected the opposite?

Actually wondering why they can't all be on the same hook? (Relevant calls only enabled if when their ids are provided anyway...)

Reply:
Yeah, that's right, they should really be around the other way. I think it ended up like this cos I was only refactoring one use case in this PR and that's how they were organised in the implementations. 

I think I'll leave this to the [other PR](#4009) as I've already made some additional changes in there.